src/domain/books/BooksPage.tsx

```diff
@@ -1,7 +1,36 @@
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { AppDispatch } from "../../store/store";
+import { booksSelector, getBooks } from "./booksSlice";
+import type { Book } from "./types";
+import { BookCard } from "../../components/BookCard";
+import "./BooksPage.css";
+
 export const BooksPage = (): JSX.Element => {
+
+  const dispatch: AppDispatch = useDispatch();
+  const { books, loading } = useSelector(booksSelector);
+
+  useEffect(() => {
+    // fetch latest books
+    dispatch(getBooks());
+  }, []);
+
   return (
-    <div>
-      Books Page
+    <div className="books-page-wrapper">
+      {loading === "loading" ? (
+          <span className="book-loading">grabbing books</span>
+        ) : (
+          <div className="books-page-container">
+            {books.map((book: Book) => (
+              <BookCard 
+                {...book} 
+                key={book.id}
+              />
+            ))}
+          </div>
+        )
+      }
```

Comment:
This is OK. But there are more than two conditions here i.e. error. Ideally, write this logic in a fn with guard clauses and call it here.

Reply:
Your JSX should be as clean as possible